modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/PostgisFilterToSQL.java

```diff
@@ -92,6 +94,21 @@ protected void visitLiteralGeometry(Literal expression) throws IOException {
     }
 
     @Override
+    protected Object evaluateLiteral(Literal expression, Class target) {
+        Date converted = null;
+        if (target == PostGISDialect.XDate.class) {
+            converted = Converters.convert(expression.evaluate(null), Date.class);
+        }
+        if (converted != null) {
+            return converted.getTime();
+        } else {
+            return super.evaluateLiteral(expression, target);
```

Comment:
As a note, in the PostgisUDTOnlineTest's testBigDates(), it always goes to this line. It never runs the XDate part of the code...